src/main.ts

```diff
@@ -1,101 +1,111 @@
+        let tenantId = secrets.getSecret("$.tenantId", false);
+        let subscriptionId = secrets.getSecret("$.subscriptionId", false);
+        let resourceManagerEndpointUrl = secrets.getSecret("$.resourceManagerEndpointUrl", false);
+        let environment = core.getInput("environment");
+        const enableAzPSSession = core.getInput('enable-AzPSSession').toLowerCase() === "true";
+        if (!servicePrincipalId || !servicePrincipalKey || !tenantId || !subscriptionId) {
+            throw new Error("Not all values are present in the creds object. Ensure clientId, clientSecret, tenantId and subscriptionId are supplied.");
+        }
+
+        // Attempting Az cli login
+        if (environment.toLowerCase() == "azurestack") {
+            if (!resourceManagerEndpointUrl) {
+                throw new Error("resourceManagerEndpointUrl is a required parameter when environment is defined.");
+            }
+            console.log(`Unregistering cloud: "${environment}" first if it exists`);
+            try {
+                await executeAzCliCommand(`cloud set -n AzureCloud`, true);
+                await executeAzCliCommand(`cloud unregister -n "${environment}"`, false);
+            } catch (error) {
+                console.log(`Ignore cloud not registered error: "${error}"`);
+            }
+            console.log(`Registering cloud: "${environment}" with ARM endpoint: "${resourceManagerEndpointUrl}"`);
+            try {
+                let baseUri = resourceManagerEndpointUrl;
+                if (baseUri.endsWith('/')) {
+                    baseUri = baseUri.substring(0, baseUri.length-1); // need to remove trailing / from resourceManagerEndpointUrl to correctly derive suffixes below
+                }
+                let suffixKeyvault = ".vault" + baseUri.substring(baseUri.indexOf('.')); // keyvault suffix starts with .
+                let suffixStorage = baseUri.substring(baseUri.indexOf('.')+1); // storage suffix starts without .
+                let profileVersion = "2019-03-01-hybrid";
+                await executeAzCliCommand(`cloud register -n "${environment}" --endpoint-resource-manager "${resourceManagerEndpointUrl}" --suffix-keyvault-dns "${suffixKeyvault}" --suffix-storage-endpoint "${suffixStorage}" --profile "${profileVersion}"`, false);
+            } catch (error) {
+                core.error(`Error while trying to register cloud "${environment}": "${error}"`);
+            }
+            await executeAzCliCommand(`cloud set -n "${environment}"`, false);
+            console.log(`Done registering cloud: "${environment}"`);
+        }
+           
+        //setting context to Azure Gov Cloud
+        if (environment.toLowerCase() == AzureCloudName.AzureUSGovernment|| environment.toLowerCase() ==AzureCloudName.AzureChinaCloud || environment.toLowerCase() ==AzureCloudName.AzureGermanCloud  || environment.toLowerCase() ==AzureCloudName.AzureCloud  ) {
```

Comment:
is there a way to check if environment is in enum? This might be better for extensibility.

Reply:
Check if this can be leveraged https://stackoverflow.com/questions/43804805/check-if-value-exists-in-enum-in-typescript

Reply:
instead of checking for toLoweCase everywhere, you can also convert the input to lower case after reading the input like:
```typescript
let environment = core.getInput("environment");
environment = environment.toLowerCase();
```

Reply:
Lets just remove the if check and directly run 
```typescript
await executeAzCliCommand(`cloud set -n "${environment}"`, false);
```.

We are also setting the cloud environment inside the AzureStack if condition, remove it from there. So the above statement will take care of all cloud environments.